app/src/org/commcare/activities/PromptActivity.java

```diff
@@ -0,0 +1,150 @@
+package org.commcare.activities;
+
+import android.content.Intent;
+import android.graphics.Typeface;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.commcare.dalvik.BuildConfig;
+import org.commcare.dalvik.R;
+import org.commcare.interfaces.PromptItem;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Activity used to prompt users to update to a newer .apk or app version, or to reinstall CommCare.
+ *
+ * Created by amstone326 on 4/19/17.
+ */
+public abstract class PromptActivity extends CommCareActivity {
+
+    protected static final int DO_AN_UPDATE = 1;
+
+    public static String FROM_RECOVERY_MEASURE = "from-recovery-measure";
+
+    protected PromptItem toPrompt;
+
+    protected TextView promptTitle;
+    protected Button actionButton;
+    protected Button doLaterButton;
+    protected ImageView imageCue;
+    protected TextView instructions;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (toPrompt == null) {
+            refreshPromptObject();
+        }
+        if (savedInstanceState == null &&
+                !getIntent().getBooleanExtra(FROM_RECOVERY_MEASURE, false)) {
+            // on initial activity load only
+            toPrompt.incrementTimesSeen();
+        }
+        setupUI();
+    }
+
+    private void setupUI() {
+        setContentView(R.layout.prompt_view);
+        promptTitle = findViewById(R.id.prompt_title);
```

Comment:
@ctsims  Since the PromptActivity is not the top level class, our annotation processing fails for this use case. We only look for the declared fields in the top level class currently and will need to modify that code to process the super classes as well if we wanna achieve this.

Reply:
ah, right. Makes sense